output/cloud/expv2/hdr.go

```diff
@@ -37,14 +38,13 @@ const (
 // The current version is: f(N = 25, m = 7) = 3200.
 type histogram struct {
 	// Buckets stores the counters for each bin of the histogram.
-	// It does not include the first and the last absolute bucket,
-	// because they contain exception cases
-	// and they requires to be tracked in a dedicated way.
-	//
-	// It is expected to start and end with a non-zero bucket,
-	// in this way we can avoid extra allocation for not significant buckets.
-	// All the zero buckets in between are preserved.
-	Buckets []uint32
+	// It does not include counters for the untrackable values,
+	// because they contain exception cases and require to be tracked in a dedicated way.
+	Buckets map[uint32]uint32
+
+	// Indexes keeps an ordered slice of unique-seen buckets' indexes.
+	// It allows to iterate the buckets in order. It uses an ascendent order.
+	Indexes []uint32
```

Comment:
We may consider using a Btree as an optimization. But I would like to achieve stability in the protocol before.

Reply:
I wonder if we can't just order the buckets indexes at the end :shrug:. There is no need to constantly keep track of it -if we only want it to be in order :shrug:

Reply:
Yeah, it simplifies the code a lot, good suggestion.